From c45721f2be939104b11d47f704108fbcfbec4a51 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Wed, 6 Nov 2019 18:32:46 +0100
Subject: [PATCH] disable static_assert on AtomicCounter::is_always_lock_free
 on armel ...

... since  AtomicCounter::is_always_lock_free seems to be
always false on armel

Change-Id: I89c15fd41337ac5b29f781722423427825e6ff45
---
 vcl/inc/opengl/zone.hxx | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vcl/inc/opengl/zone.hxx b/vcl/inc/opengl/zone.hxx
index 3210186c3096..13ac3bf6793e 100644
--- a/vcl/inc/opengl/zone.hxx
+++ b/vcl/inc/opengl/zone.hxx
@@ -36,7 +36,9 @@ class VCL_DLLPUBLIC OpenGLZone {
     // increasing, so will eventually overflow, so the underlying type better be unsigned, which
     // sig_atomic_t is not guaranteed to be:
     using AtomicCounter = std::atomic<std::make_unsigned_t<std::sig_atomic_t>>;
+#if !defined ARM32 || (defined ARM32 && defined __ARM_PCS_VFP)
     static_assert(AtomicCounter::is_always_lock_free);
+#endif
 
     /// how many times have we entered a GL zone
     static AtomicCounter gnEnterCount;
-- 
2.20.1

